src/dense/stiff_addsteps.jl

```diff
@@ -27,33 +28,28 @@ function DiffEqBase.addsteps!(k,t,uprev,u,dt,f,p,cache::Union{Rosenbrock23Cache,
 
     # Assignments
     sizeu  = size(u)
-    #mass_matrix = integrator.f.mass_matrix
+    mass_matrix = f.mass_matrix
     γ = dt*d
     dto2 = dt/2
     dto6 = dt/6
 
-    #@.. linsolve_tmp = @muladd fsalfirst + γ*dT
-    @tight_loop_macros for i in uidx
-      @inbounds linsolve_tmp[i] = @muladd fsalfirst[i] + γ*dT[i]
-    end
+    @.. linsolve_tmp = @muladd fsalfirst + γ*dT
 
     ### Jacobian does not need to be re-evaluated after an event
     ### Since it's unchanged
-    for i in 1:length(u), j in 1:length(u)
-      @inbounds W[i,j] = @muladd I[i,j]-γ*J[i,j]
-    end
+    jacobian2W!(W, mass_matrix, γ, J, false)
```

Comment:
@YingboMa this might change the mass matrix sensitivity results since now MM is involved in the `addsteps!`